### PR TITLE
Fixing Fast Registration for non root dirs

### PR DESCRIPTION
**Pull request by kumare3**
Signed-off-by: Ketan Umare <ketan.umare@gmail.com>

# TL;DR
In the case when using a container with a non root user, the inflation dir for the code has to be specified. It is common practice to skip a trailing slash "/mypath`/`". so when the distribution is downloaded, it will instead write this to this filename
And this fails the tar logic, which loooks for a similarly named file under the destination path.

The solution simply enforces that the given destination path is actually a directory

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
https://github.com/flyteorg/flyte/issues/3022

### CODE REVIEW

**Review by wild-endeavor — DISMISSED** [review]

**Review by pingsutw — APPROVED** [review]

File: flytekit/tools/fast_registration.py [hunk 1]
```diff
@@ -105,10 +103,14 @@ def download_distribution(additional_distribution: str, destination: str):
     :param Text additional_distribution:
     :param os.PathLike destination:
     """
-    file_access.get_data(additional_distribution, destination)
+    if not destination:
+        raise ValueError("Destination path is required to download distribution and it should be a directory")
```
Comment (eapolinario) [inline flytekit/tools/fast_registration.py:107]: should we use https://docs.python.org/3/library/os.path.html#os.path.isdir and raise if it returns false?

File: flytekit/tools/fast_registration.py [hunk 2]
```diff
@@ -105,10 +103,14 @@ def download_distribution(additional_distribution: str, destination: str):
...
+    # NOTE the os.path.join(destitation, ''). This is to ensure that the given path is infact a directory and all
```
Comment (eapolinario) [inline flytekit/tools/fast_registration.py:108]: typo: destitation.

File: flytekit/tools/fast_registration.py [hunk 3]
```diff
@@ -105,10 +103,14 @@ def download_distribution(additional_distribution: str, destination: str):
...
+    # downloaded data should be copied into this directory
+    FlyteContextManager.current_context().file_access.get_data(additional_distribution, os.path.join(destination, ""))
```
Comment (eapolinario) [inline flytekit/tools/fast_registration.py:110]: Can you also leave a comment saying that we're doing this because of `fsspec` ?

**Review by eapolinario — APPROVED** [review]
